platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/binding/post/PostResponseCreator.java

```diff
@@ -79,7 +79,10 @@ public Response getSamlpResponse(
     submitFormUpdated = submitFormUpdated.replace("{{" + Idp.SAML_TYPE + "}}", "SAMLResponse");
     submitFormUpdated =
         submitFormUpdated.replace("{{" + Idp.SAML_RESPONSE + "}}", encodedSamlResponse);
-    submitFormUpdated = submitFormUpdated.replace("{{" + Idp.RELAY_STATE + "}}", relayState);
+    submitFormUpdated =
+        relayState != null
+            ? submitFormUpdated.replace("{{" + Idp.RELAY_STATE + "}}", relayState)
+            : submitFormUpdated.replace("{{" + Idp.RELAY_STATE + "}}", "");
```

Comment:
✏️ Could also do 
```java
submitFormUpdated =
  submitFormUpdated.replace(String.format("{{%s}}", Idp.RELAY_STATE), relayState != null ? relayState : "");
```

Reply:
If performance matters shouldn't we stick with concatenation?